libjulia/optimiser/MainFunction.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * Changes the topmost block to be a function with a specific name ("main") which has no
+ * inputs nor outputs.
+ */
+
+#include <libjulia/optimiser/MainFunction.h>
+
+#include <libsolidity/inlineasm/AsmData.h>
+
+#include <libsolidity/interface/Exceptions.h>
+
+#include <libdevcore/CommonData.h>
+
+using namespace std;
+using namespace dev;
+using namespace dev::julia;
+using namespace dev::solidity;
+
+void MainFunction::operator()(Block& _block)
+{
+	solAssert(_block.statements.size() >= 1, "");
+	solAssert(_block.statements[0].type() == typeid(Block), "");
+
+	Block & block = boost::get<Block>(_block.statements[0]);
+	_block.statements[0] = FunctionDefinition{
+		block.location,
+		"main",
+		{},
+		{},
+		std::move(_block)
```

Comment:
Hm, this is a typo, it should be `block` and not `_block`. Why does the tests succeed?

Reply:
yeah, weird. This should even cause segmentation faults.